### PR TITLE
First problem

### PR DESCRIPTION
**Pull request by HiciuStefan**

### CODE REVIEW

File: first_problem_test.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require 'test/unit'
+require_relative'./first_problem'
+
+class FirstProblemTest < Test::Unit::TestCase
+
+  def test_compute_empty_array_0
+    array = Array.new
+    assert_equal(-1,FirstProblem.new(array).compute)
```
Comment (alexandru-calinoiu) [inline first_problem_test.rb]: Not very good at number theory but shouldn't this be `0`?

Comment (HiciuStefan) [reply first_problem_test.rb]: i was trying to push for a "refactor" kind of step. Seems rather weird ideea now. Don't know exactly what i was thinking
But yes , it's 0...

File: first_problem.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  def initialize(input_array)
+    @numbers_array = input_array
+  end
+
+  def compute()
+    sum = 0
+    @numbers_array.each do |number|
+      sum += number if number.even?
+    end
+    sum
```
Comment (alexandru-calinoiu) [inline first_problem.rb]: The `sum` lines in this code look really strange, are you familiar with the `Enumerator` concept? Can you express this as a concatenation of operations?

File: first_problem_test.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+require 'test/unit'
+require_relative'./first_problem'
+
+class FirstProblemTest < Test::Unit::TestCase
+
+  def test_compute_empty_array_0
+    array = Array.new
```
Comment (alexandru-calinoiu) [inline first_problem_test.rb:7]: Usually unit tests have 3 regions: `arrange`, `execute` and `assert`, is usually a good idea to have those show in the test.

(nothing to do for now just sharing insights)

File: first_problem.rb (hunk 2 above)
Comment (HiciuStefan) [reply first_problem.rb]: updated code

File: first_problem.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+class FirstProblem
+
+
+  def initialize(input_array)
+    @numbers_array = input_array
+  end
+
+  def compute()
+    sum = 0
+    @numbers_array.each {|number| sum+=number if number.even? }
```
Comment (alexandru-calinoiu) [inline first_problem.rb:10]: I would have go with something like:

```
numbers.select(&:even?).inject(:+)
```

**Comment by alexandru-calinoiu** [comment]
Thank you, will write a email shortly.

**Pull request closed without merging**